packages/babel-parser/src/parser/expression.js

```diff
@@ -67,14 +67,13 @@ export default class ExpressionParser extends LValParser {
   +parseFunctionParams: (node: N.Function, allowModifiers?: boolean) => void;
   +takeDecorators: (node: N.HasDecorators) => void;
 
-  // Check if property name clashes with already added.
-  // Object/class getters and setters are not allowed to clash —
-  // either with each other or with an init property — and in
-  // strict mode, init properties are also not allowed to be repeated.
+  // Check if property __proto__ has been used more than once.
+  // If the expression is a destructuring assignment, then __proto__ may appear
+  // multiple times. Otherwise, __proto__ is a duplicated key.
 
-  checkPropClash(
+  checkDuplicatedProto(
     prop: N.ObjectMember | N.SpreadElement,
-    propHash: { [key: string]: boolean },
+    propHash: { [key: string]: number },
```

Comment:
Could you rename this to something like `protoRef`?
Also, ideally it's type should be `{ used: boolean, start?: number }`.
And then, when we find `__proto__`,
1. If `used` is true, set `start`
2. Set `used` to true

Reply:
Thanks, @nicolo-ribaudo. Wouldn't there be an issue if we had more than two occurences of `__proto__`? In this way, `start` would be the last occurrence's start position instead of the first duplicated one. It would be an error all the same if it happens outside of a destructuring pattern but it wouldn't give us the absolute first error position. Does that matter in this case?

Reply:
> In this way, start would be the last occurrence's start position instead of the first duplicated one.

Ok this is the best behavior. We can do `if (protoRef.used && !protoRef.start)` to preserve it.